### ISSUE TITLE
How to modify code for single-variate time series?

### ISSUE THREAD

**Issue by Shubhamsg17** [gzerveas/mvts_transformer#51]

**Comment by gzerveas** [comment]
Hi, probably the easiest way to make it work would be the following hack:

In your dataset class inside `data.py`, replicate the single column that currently defines your single variable into one or more copies, in a way that you end up with e.g. 3 identical columns. You won't need to change anything else in the code for this.

An alternative, more nuanced way would include defining a single 1D-convolutional layer with M filters (M is a hyperparameter) such that you end up with M features per time step. This way you could also do "smart down-sampling" (feature extraction) in case your temporal resolution is very high and results in a large number of time steps.

So basically right after your input `X` you would only need to add a layer like this:

```python
out_channels = M
kernel_size = 1
stride = 1
conv = torch.nn.Conv1d(in_channels=1, out_channels=out_channels, kernel_size=kernel_size, stride=stride)
X = conv(X)
```

The above doesn't do any feature extraction/down-sampling, so if you do want this, then you can set `stride` and `kernel_size` as hyperparameters to larger values.
